### ISSUE TITLE
unvalid results of demo

### ISSUE THREAD

**Issue by SuhZhang** [vislearn/ngransac#25]
Hi :) I tried to test NG-RANSAC on YFCC100M using SIFT, but got 0 AUC unexpectedly. When I am looking for the problem and replicating the demo.py by running `python ngransac_demo.py -img1 images/demo1.jpg -fl1 900 -img2 images/demo2.jpg -fl2 900,` the results seem unvalid for almost all correspondences concentrate to the same area. Here is the image:

![Image](https://github.com/user-attachments/assets/785aac92-ee84-4aef-b993-a2e41c66189f)

The envs:
python    3.6.15
opencv-contrib-python    3.4.11.45
pytorch    1.10.1
CUDA    11.3

**Comment by ebrach** [comment]
Hi,

please verify that you are using OpenCV version 3.4.2 as stated in the readme.

Best,
Eric

**Comment by SuhZhang** [comment]
Thx for you reply. I have tried to swtich OpenCV to 3.4.2.17 and pytorch to 1.2.0 but got the same unvalid results. However, the problem was fixed when useing -nos which ignores the side infomation.

![Image](https://github.com/user-attachments/assets/da25ca50-89cc-4fab-ad83-74bf22f52f86)

Tell me if I'm messing up. Greatly appreciate.

**Comment by ebrach** [comment]
Hmm. This is unexpected. Can you please post your console output when running the demo both with and without the -nos flag?

**Comment by SuhZhang** [comment]
python ngransac_demo.py -img1 images/demo1.jpg -fl1 900 -img2 images/demo2.jpg -fl2 900

Fitting Essential Matrix...

Using SIFT.

No model file specified. Inferring pre-trained model from given parameters:
models/weights_e2e_E_r1.00_.net
Successfully loaded model.

Processing pair:
Image 1:  images/demo1.jpg
Image 2:  images/demo2.jpg

Feature found in image 1: 2000
Feature found in image 2: 2000

Number of valid matches: 2000

=== Model found by RANSAC: ==========

[[-0.55587167 -0.4306436   0.05926346]
 [ 0.41969678 -0.55434184 -0.12223194]
 [-0.01167158 -0.05927362 -0.00318125]]

RANSAC Inliers: 188

=== Model found by NG-RANSAC: =======

[[ 0.03526158 -0.70331156 -0.04548426]
 [ 0.5893719   0.00700524  0.3885519 ]
 [ 0.03592258 -0.044696    0.01927547]]

NG-RANSAC Inliers:  190

Done. Visualization of the result stored as demo.png

![Image](https://github.com/user-attachments/assets/29ad34d4-dd7c-48a4-8c0b-585f91436c5c)


python ngransac_demo.py -img1 images/demo1.jpg -fl1 900 -img2 images/demo2.jpg -fl2 900 -nos

Fitting Essential Matrix...

Using SIFT.

No model file specified. Inferring pre-trained model from given parameters:
models/weights_e2e_E_r1.00_.net
Successfully loaded model.

Processing pair:
Image 1:  images/demo1.jpg
Image 2:  images/demo2.jpg

Feature found in image 1: 2000
Feature found in image 2: 2000

Number of valid matches: 2000

=== Model found by RANSAC: ==========

[[-0.55587167 -0.4306436   0.05926346]
 [ 0.41969678 -0.55434184 -0.12223194]
 [-0.01167158 -0.05927362 -0.00318125]]

RANSAC Inliers: 188

=== Model found by NG-RANSAC: =======

[[-0.01493459 -0.6780253   0.00610476]
 [ 0.6881042  -0.00657859  0.16088963]
 [-0.02012288  0.20035848 -0.00753758]]

NG-RANSAC Inliers:  118

Done. Visualization of the result stored as demo.png

![Image](https://github.com/user-attachments/assets/92f792f7-757d-469d-b409-801be24de462)

**Comment by ebrach** [comment]
Thank you. Can you also run a quantitative analysis as suggested in response to the following issue? https://github.com/vislearn/ngransac/issues/2#issue-497384008

**Comment by SuhZhang** [comment]
The quantitative analysis results perfectly match what you reported.


python ngransac_test.py -ds reichstag

No model file specified. Inferring pre-trained model from given parameters:
models/weights_e2e_E_r1.00_.net
Successfully loaded model.
Starting evaluation for dataset: reichstag test_data 

Processing batch 1 of 2
Processing batch 2 of 2

Avg. Model Time: 9ms
Avg. RANSAC Time: 21ms

=== Relative Pose Accuracy ===========================
AUC for 5deg/10deg/20deg: 0.44/0.57/0.64



python ngransac_test.py -ds sacre_coeur

No model file specified. Inferring pre-trained model from given parameters:
models/weights_e2e_E_r1.00_.net
Successfully loaded model.
Starting evaluation for dataset: sacre_coeur test_data 

Processing batch 1 of 32
Processing batch 2 of 32
Processing batch 3 of 32
Processing batch 4 of 32
Processing batch 5 of 32
Processing batch 6 of 32
Processing batch 7 of 32
Processing batch 8 of 32
Processing batch 9 of 32
Processing batch 10 of 32
Processing batch 11 of 32
Processing batch 12 of 32
Processing batch 13 of 32
Processing batch 14 of 32
Processing batch 15 of 32
Processing batch 16 of 32
Processing batch 17 of 32
Processing batch 18 of 32
Processing batch 19 of 32
Processing batch 20 of 32
Processing batch 21 of 32
Processing batch 22 of 32
Processing batch 23 of 32
Processing batch 24 of 32
Processing batch 25 of 32
Processing batch 26 of 32
Processing batch 27 of 32
Processing batch 28 of 32
Processing batch 29 of 32
Processing batch 30 of 32
Processing batch 31 of 32
Processing batch 32 of 32

Avg. Model Time: 1ms
Avg. RANSAC Time: 19ms

=== Relative Pose Accuracy ===========================
AUC for 5deg/10deg/20deg: 0.52/0.56/0.61

**Comment by ebrach** [comment]
Ok, then I think there is no problem, and the demo results are due to randomness as explained in issue https://github.com/vislearn/ngransac/issues/2#issue-497384008.

Your problems on YFCC100M might be due to incorrect data preparation? If you overwrite the NG-RANSAC probabilities with ones (e.g. by using the -uniform flag in test_ngransac.py) you can rule out model failure.

If you do that, and still get zero AUC (basically when using SIFT with standard RANSAC), something is not right with how the errors are calculated wrt the dataset ground truth.

**Comment by ebrach** [comment]
Closing for now. Feel free to re-open if you have more questions.
